src/librustc_error_codes/error_codes/E0688.md

```diff
@@ -0,0 +1,36 @@
+In-band lifetimes has mixed with explicit lifetime binders.
+
+Erroneous code example:
+
+```compile_fail,E0688
+#![feature(in_band_lifetimes)]
+
+fn foo<'a>(x: &'a u32, y: &'b u32) {}   // error!
+
+struct Foo<'a> { x: &'a u32 }
+
+impl Foo<'a> {
+    fn bar<'b>(x: &'a u32, y: &'b u32, z: &'c u32) {}   // error!
+}
+
+impl<'b> Foo<'a> {  // error!
+    fn baz() {}
+}
+```
+
+We cannot mix in-band lifetimes with explicit lifetime binders.
```

Comment:
```suggestion
In-band lifetimes cannot be mixed with explicit lifetime binders.
```

This is really just a re-phrasing of the error, it'd be great to elaborate a little - something like "a function/struct/whatever can either have explicit lifetime binders *or* in-band lifetimes".